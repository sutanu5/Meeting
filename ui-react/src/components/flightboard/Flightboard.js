import React, { Component } from 'react';
import PropTypes from 'prop-types';

import FlightboardRow from './FlightboardRow';
import Socket from '../global/Socket';
import Spinner from '../global/Spinner';

class Flightboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      response: false,
      error: false,
      now: new Date(),
      rooms: []
    }

    this.handleSocket = this.handleSocket.bind(this);
  }

  getRoomData() {
    return fetch('/api/rooms')
      .then((response) => response.json())
      .then((data) => {
        if(!data.error){
          this.setState({
            response: true,
            error: false,
            rooms: data
          });
        }
        else {
          this.setState({
            response: true,
            error: true,
            rooms: data
          });
        }
      });
  }

  handleSocket(socketResponse) {
    this.setState({
      response: socketResponse.response,
      now: socketResponse.now,
      rooms: socketResponse.rooms
    });
  }

  componentDidMount = () => {
    this.getRoomData();
  }

  render() {
    const { error, now, response, rooms } = this.state;

    return (
      <div className="tracker-wrap">
        <Socket response={this.handleSocket} />

        { response ?
          (!error ?
            rooms.map((room, key) =>
              <FlightboardRow room={room} key={key} filter={this.props.filter} />
            )
          :
            <div className="container">
              <div className="credentials-error">{rooms.error}</div>
            </div>
          )
        :
          <Spinner />
        }
      </div>
    );
  }

}

Flightboard.propTypes = {
  filter: PropTypes.string
};

export default Flightboard;
