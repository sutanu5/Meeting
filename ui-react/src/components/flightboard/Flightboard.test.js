import Flightboard from './Flightboard';

describe('Flightboard component', () => {

  beforeEach(() => {
    Flightboard.prototype.getRoomData = jest.fn();
    Flightboard.prototype.handleSocket = jest.fn();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Flightboard />);

    expect(wrapper).toMatchSnapshot();
  });

  it('calls componentDidMount', () => {

  });

  it('calls getRoomData', () => {
    const wrapper = shallow(<Flightboard />); 

    const instance = wrapper.instance();
    const getRoomData = jest.spyOn(instance, 'getRoomData');
    
    // instance.getRoomData();
    expect(getRoomData).toHaveBeenCalled();
  });

  it('calls handleSocket', () => {
    const wrapper = shallow(<Flightboard />);

    const instance = wrapper.instance();
    const handleSocket = jest.spyOn(instance, 'handleSocket');
    
    instance.handleSocket('foo');
    expect(handleSocket).toHaveBeenCalledWith('foo');
  });

  it('renders Spinner component if response is false', () => {
    const wrapper = shallow(<Flightboard />);

    const spinner = wrapper.find('Spinner').exists();
    expect(spinner).toBeTruthy();
  });

  it('renders credentials error if response is true and error is true', () => {
    const wrapper = shallow(<Flightboard />);
    wrapper.setState({ 
      response: true, 
      error: true,
      rooms: {
        'error':'Credentials error'
      }
    });

    const error = wrapper.find('.credentials-error').text();
    expect(error).toBe(wrapper.state().rooms.error);
  });

  // it('renders credentials error if response is true and error is true', () => {
  //   const wrapper = shallow(<Flightboard />);
  //   wrapper.state({ 
  //     response: true, 
  //     error: true,
  //     rooms: [{
  //       'Roomlist':'Test Roomlist',
  //       'Name':'Test Room',
  //       'RoomAlias':'test-room',
  //       'Email':'email@email.com',
  //       'Appointments':[
  //          {
  //             'Subject':'Meeting Subject',
  //             'Organizer':'John Doe',
  //             'Start':1532966400000,
  //             'End':1533344400000
  //          }
  //       ],
  //       'Busy':true
  //     }]
  //   });

  //   const error = wrapper.find('.credentials-error').exists();
  //   expect(error).not.toBeTruthy();
  // });
});