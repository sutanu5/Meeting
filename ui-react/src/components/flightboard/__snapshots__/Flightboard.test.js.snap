// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flightboard component renders correctly 1`] = `
<div
  className="tracker-wrap"
>
  <Socket
    response={[Function]}
  />
  <Spinner />
</div>
`;
